src/managed/Microsoft.NET.HostModel/AppHost/MachOUtils.cs

```diff
@@ -0,0 +1,310 @@
+        ///      * SYMTAB 
+        ///      * (Some alignment bytes)
+        ///      * The Code-signature
+        ///      
+        /// In order to remove the signature, the method:
+        /// - Removes (zeros out) the LC_CODE_SIGNATURE command
+        /// - Adjusts the size and count of the load commands in the header
+        /// - Truncates the size of the __LINKEDIT segment to the end of SYMTAB
+        /// - Truncates the apphost file to the end of the __LINKEDIT segment
+        /// 
+        /// </summary>
+        /// <param name="filePath">Path to the AppHost</param>
+        /// <returns>
+        ///  True if 
+        ///    - The input is a MachO binary, and 
+        ///    - It is a signed binary, and 
+        ///    - The signature was successfully removed
+        ///   False otherwise 
+        /// </returns>
+        /// <exception cref="AppHostMachOFormatException">
+        /// The input is a MachO file, but doesn't match the expect format of the AppHost.
+        /// </exception>
+        unsafe public static bool RemoveSignature(string filePath)
+        {
+            using (var stream = new FileStream(filePath, FileMode.Open, FileAccess.ReadWrite))
+            {
+                uint signatureSize = 0;
+                using (var mappedFile = MemoryMappedFile.CreateFromFile(stream,
+                                                                        mapName: null,
+                                                                        capacity: 0,
+                                                                        MemoryMappedFileAccess.ReadWrite,
+                                                                        HandleInheritability.None,
+                                                                        leaveOpen: true))
+                {
+                    using (var accessor = mappedFile.CreateViewAccessor())
+                    {
+                        byte* file = null;
+                        RuntimeHelpers.PrepareConstrainedRegions();
+                        try
+                        {
+                            accessor.SafeMemoryMappedViewHandle.AcquirePointer(ref file);
+                            Verify(file != null, MachOFormatError.MemoryMapAccessFault);
+
+                            MachHeader* header = (MachHeader*)file;
+
+                            if (!header->IsValid())
+                            {
+                                // Not a MachO file.
+                                return false;
+                            }
+
+                            Verify(header->Is64BitExecutable(), MachOFormatError.Not64BitExe);
+
+                            file += sizeof(MachHeader);
+                            SegmentCommand64* linkEdit = null;
+                            SymtabCommand* symtab = null;
+                            LinkEditDataCommand* signature = null;
+
+                            for (int i = 0; i < header->ncmds; i++)
```

Comment:
Might want to declare `i` as `uint` too (`ncmds` is also unsigned).

Reply:
Fixed this too.